cg/meta/clean/flow_cell_run_directories.py

```diff
@@ -0,0 +1,102 @@
+        self.flow_cell_name: str = self.flow_cell_dir.name
+        self.identifier: str = self.flow_cell_name.split(UNDERSCORE)[FLOW_CELL_IDENTIFIER_POSITION]
+        self.run_date: str = self.flow_cell_name.split(UNDERSCORE)[FLOW_CELL_DATE_POSITION]
+        self.derived_date: datetime = datetime.strptime(self.run_date, "%y%m%d")
+        self.id: str = self.identifier[1:]
+        self._age: Optional[timedelta] = None
+        self._sequenced_date: Optional[datetime] = None
+        self.sample_sheet_path: Path = (
+            self.flow_cell_dir / DemultiplexingDirsAndFiles.SAMPLE_SHEET_FILE_NAME
+        )
+
+    @property
+    def age(self) -> int:
+        """How long ago a flow cell was sequenced, in days"""
+        if self._age is None:
+            LOG.info("Setting age property")
+            self._age = (datetime.now() - self.sequenced_date).days
+        return self._age
+
+    @property
+    def sequenced_date(self) -> datetime:
+        """The date on which the flow cell was sequenced"""
+        if self._sequenced_date is None:
+            LOG.info("Setting sequenced date property")
+            if self.status_db.flowcell(name=self.id):
+                LOG.info("Found flow cell %s in statusdb, getting sequenced date.", self.id)
+                self._sequenced_date = self.status_db.flowcell(name=self.id).sequenced_at
+            else:
+                LOG.info(
+                    "Flow cell %s NOT found in statusdb, deriving sequenced date from run dir name!",
+                    self.id,
+                )
+                self._sequenced_date = self.derived_date
+
+        return self._sequenced_date
+
+    def remove_run_directory(self):
+        """Removes the flow cell run directory"""
+        LOG.info("Removing run directory %s", self.flow_cell_dir)
+        shutil.rmtree(self.flow_cell_dir, ignore_errors=True)
+
+    def archive_sample_sheet(self):
+        """Archives a sample sheet in housekeeper-bundles"""
+        LOG.info("Start archiving sample sheet %s", self.sample_sheet_path)
+        if not self.sample_sheet_path.exists():
+            LOG.warning("Sample sheet does not exists!")
+            return
+        LOG.info("Sample sheet found!")
+        hk_bundle: hk_models.Bundle = self.hk.bundle(name=self.id)
+        if hk_bundle is None:
+            LOG.info("Creating bundle with name %s", self.id)
+            hk_bundle = self.hk.create_new_bundle_and_version(name=self.id)
+
+        with self.hk.session_no_autoflush():
+            hk_version: hk_models.Version = self.hk.last_version(bundle=hk_bundle.name)
+            if self.hk.files(path=str(self.sample_sheet_path)).first() is None:
+                LOG.info(f"Adding sample sheet to Housekeeper")
+                tags: List[str] = [HousekeeperTags.ARCHIVED_SAMPLE_SHEET, self.id]
+                self.hk.add_file(
+                    path=str(self.sample_sheet_path), version_obj=hk_version, tags=tags
+                )
+        self.hk.commit()
```

Comment:
This code is awfully similiar to: 
```
   def add_sample_sheet_to_housekeeper(self, sample_sheet_path: Path):
        """Adds an archive sample sheet to Housekeeper"""

        hk_bundle: hk_models.Bundle = self.hk.bundle(self.id)
        if hk_bundle is None:
            hk_bundle = self.hk.create_new_bundle_and_version(name=self.id)

        with self.hk.session_no_autoflush():
            hk_version: hk_models.Version = self.hk.last_version(bundle=hk_bundle.name)
            if self.hk.files(path=str(sample_sheet_path)).first() is None:
                LOG.info(f"Adding archived sample sheet: {str(sample_sheet_path)}")
                tags: List[str] = [HousekeeperTags.ARCHIVED_SAMPLE_SHEET, self.id]
                self.hk.add_file(path=str(sample_sheet_path), version_obj=hk_version, tags=tags)
        self.hk.commit()
```
in cg/meta/clean/demultiplexed_flow_cells.py. Could this be move to HouseKeeperAPI to be used by both these classes. It seems unnecessary to repeat this code block in multiple places

Reply:
I removed this entirely and used Housekeeper's `add_and_include_file_to_latest_version` method instead, which I somehow missed before.

in PR https://github.com/Clinical-Genomics/cg/pull/1334, it was initially agreed upon that the sample sheets should be archived in a separate directory, but now the requirement is to include those sample sheets instead. I can change that too, but I'm reluctant to do that in this PR because it will require time consuming retesting on stage.

Reply:
Agreed!